### ISSUE TITLE
Crash on iOS

### ISSUE THREAD

**Issue by AGulev** [defold/editor2-issues#2908]
I got the following message after rejected app review:

_Guideline 2.1 - Performance - App Completeness

We were unable to review your app as it crashed on launch. We have attached detailed crash logs to help troubleshoot this issue._

Unfortunately, I can't symbolicate their crash logs because of this issue.

But we were able to reproduce one crash (I hope it's the same) on one of the devices by following steps:

Open game
Wait while the screen will be blocked by OS
Unblock screen
I got a crash log:

![image](https://user-images.githubusercontent.com/2209596/74085460-64bfa180-4a79-11ea-9ed4-f396f95901ba.png)

**Comment by britzl** [comment]
Does the project contain any native extensions?

Which Defold version were you using?

**Comment by AGulev** [comment]
The crash happens in IAP module, this line https://github.com/defold/extension-iap/blob/8d9ea79d7ddef8bc94035f510907008db938c233/extension-iap/src/iap_ios.mm#L122
I don't know why, but I'm trying to go deeper and debug this code.

**Comment by AGulev** [comment]
I don't know why, but sometimes I get empty values for some of in-app purchases 
<img width="421" alt="image" src="https://user-images.githubusercontent.com/2209596/75825807-69d8ed80-5da6-11ea-8b21-0a60cdaad34c.png">

**Comment by AGulev** [comment]
@britzl 
After a few renaming of purchases (yes, just rename purchase in App Store Connect, and then rename it back),  I start receiving valid purchase info. 
I have no idea what it was, but now it works.
Not sure, maybe we need to modify IAP Extention and check that p.localizedTitle is not nil? (I close this ticket)

**Comment by britzl** [comment]
> Not sure, maybe we need to modify IAP Extention and check that p.localizedTitle is not nil? (I close this ticket)

Yes, that might be best! Can you please create a ticket?

**Comment by AGulev** [comment]
@britzl yes, ofc

**Comment by AGulev** [comment]
@britzl it seems like I mixed up two different crashes: 
https://forum.defold.com/t/ios-crash-on-app-focus-gain-in-dmgraphics-clear-dmgraphics-context-unsigned-int-unsigned-char-unsigned-char-unsigned-char-unsigned-char-float-unsigned-int/64609

**Comment by britzl** [comment]
Yeah, there's two different and unrelated crashes in this issue. The IAP crash has been resolved: https://github.com/defold/extension-iap/issues/18

**Comment by britzl** [comment]
Also discussed here: https://forum.defold.com/t/ios-crash-on-app-focus-gain-in-dmgraphics-clear-dmgraphics-context-unsigned-int-unsigned-char-unsigned-char-unsigned-char-unsigned-char-float-unsigned-int/64609
